pkg/proxy/iptables/proxier.go

```diff
@@ -1302,13 +1302,9 @@ func (proxier *Proxier) syncProxyRules() {
 		// Capture nodeports.  If we had more than 2 rules it might be
 		// worthwhile to make a new per-service chain for nodeport rules, but
 		// with just 2 rules it ends up being a waste and a cognitive burden.
-		if svcInfo.NodePort() != 0 {
+		if svcInfo.NodePort() != 0 && len(nodeAddresses) != 0 {
```

Comment:
so you omit the nodeport rules but you keep the loadbalancer ones, should an error on node addresses stop adding the loadbalancer rules too?

Reply:
Good question.  On some platforms NodePorts are needed to implement LBs.  On other platforms not.  Should we respect `AllocateLoadBalancerNodePorts` here (that's beta) or maybe wait for "IPMode" (abandoned KEP) and fail them together?

Reply:
The loadbalancer rules are only created if you have the kind of loadbalancer that doesn't use the nodeports.

Kube-proxy should probably fail at startup time if you passed a `--nodeport-addresses` that doesn't match any addresses on the node.

After that, the only reasons `GetNodeAddresses()` could fail are: (1) the set of local IPs changed and now there is nothing left that still matches `--nodeport-addresses`, or (2) errors from `net.InterfaceAddrs()` that are basically "can't happen". In the latter case it doesn't matter much what we do because the system is probably in really bad shape anyway. In the former case, just not writing out NodePort rules seems like a plausible behavior (after logging an error); the user said "only create NodePort rules for the following IPs", and none of those IPs exist, so we create NodePort rules for no IPs...

Reply:
In the case of a proxy-ish LB which forwards to a NodePort, not installing NP rules could be a real problem, but you're right that there's not much we can do - we can't turn off the upstream LB and stop the flow of packets.  The LB rules here are moot.